dali/test/python/test_utils_tensorflow.py

```diff
@@ -48,7 +48,32 @@ def available_gpus():
     return devices
 
 
-def get_pipeline(batch_size, num_threads, device, device_id=0, shard_id=0, num_shards=1):
+# ################################################################################################ #
+#
+# To test custom DALI pipeline and DALIDataset wrapper for it all the `run_tf_dataset_*`
+# routines accept two arguments:
+#   * get_pipeline_desc
+#   * to_dataset
+# Both are callbacks, examples of those are respectively:
+#   * get_image_pipeline
+#   * to_image_dataset
+#
+# Respective signatures:
+# get_pipeline_desc(batch_size, num_threads, device, device_id, shard_id, num_shards,
+#                  def_for_dataset) -> nvidia.dali.pipeline, shapes, dtypes
+# `def_for_dataset` - indicates if this function is invoked to create a basaline standalone
+#                     pipeline (False), or it will be wrapped into TF dataset (True)
+# It is supposed to return a tuple that also describes the shapes and dtypes returned by the pipe.
+#
+#
+# to_image_dataset(image_pipeline_desc, device_str) -> tf.data.Dataset
+# image_pipeline_desc will be the tuple returned by the `get_pipeline_desc`, device_str
+# is the expected placement of the tested DALIDataset
+#
+# ################################################################################################ #
+
+def get_image_pipeline(batch_size, num_threads, device, device_id=0, shard_id=0, num_shards=1,
+        def_for_dataset=False):
```

Comment:
It's hard for me to grasp the meaning of `def_for_dataset` without an implementation.

Reply:
I found a need to have a slightly different definition for a pipeline that is run standalone vs the one used in TF (namely the need to do:
```
fn.external_source(name="...") # tf-wrapped one
fn.external_source(source=...) # DALI-only one
```
I could probably introduce explicit feed-inputs step as well, but this is easier and probably more generic.